feat(core): Move `DefaultIterableDiffer` to private API as `ɵDefaultIterableDiffer`

JeanMeche opened the pull request:
Breaking change: The public `DefaultIterableDiffer` was deprecated in v4. This commit moves `DefaultIterableDiffer` to the private space and provides a migration.

## Does this PR introduce a breaking change?

- [x] Yes

AndrewKushnir reviewed (commented):
@JeanMeche thanks for the PR 👍 

Just a couple of quick drive-by comments:

- It'd be great to split the change into 2 commits: 1st one adds the `ɵ` symbol, the 2nd commit - adds a migration.
- Since the symbol was deprecated in v4 and it's unlikely that the symbol was heavily used in application code (I found 0 usages in Google's codebase), I think that adding a migration for this change might be  excessive. We can probably just cover this in the breaking changes note.
 
Let's keep the migration for now and see what other reviewers think.

JeanMeche committed:
feat(core): Move DefaultIterableDiffer to private API as `ɵDefaultIterableDiffer`

Breaking change: The public `DefaultIterableDiffer` was deprecated in v4.
This commit moves the `DefaultIterableDiffer` to the private space

JeanMeche committed:
feat(migrations): Add a migration for DefaultIterableDiffer

`DefaultIterableDiffer` was moved from the public API to the private API. This migration will convert imports of  `DefaultIterableDiffer` to `ɵDefaultIterableDiffer as DefaultIterableDiffer`

JeanMeche commented:
> * It'd be great to split the change into 2 commits: 1st one adds the `ɵ` symbol, the 2nd commit - adds a migration.
> * Since the symbol was deprecated in v4 and it's unlikely that the symbol was heavily used in application code (I found 0 usages in Google's codebase), I think that adding a migration for this change might be  excessive. We can probably just cover this in the breaking changes note.
> 
> Let's keep the migration for now and see what other reviewers think.

Split in 2 commits is done. 2nd commit is ready to be dropped if others feel like you 👍

That's cool, I thought the move hasn't been done because it required writting a migration. 

The `update-to-version-16.md` hasn't been created yet. Is there an existing process for creating this file ?
